homeassistant/components/dwd_weather_warnings/config_flow.py

```diff
@@ -26,27 +28,58 @@ async def async_step_user(
         errors: dict = {}
 
         if user_input is not None:
-            region_identifier = user_input[CONF_REGION_IDENTIFIER]
+            valid_options = (CONF_REGION_IDENTIFIER, CONF_GPS_TRACKER)
 
-            # Validate region identifier using the API
-            if not await self.hass.async_add_executor_job(
-                DwdWeatherWarningsAPI, region_identifier
-            ):
-                errors["base"] = "invalid_identifier"
+            # Check, if either CONF_REGION_IDENTIFIER or CONF_GPS_TRACKER has been set.
+            if all(k not in user_input for k in valid_options):
+                errors["base"] = "no_identifier"
+            elif all(k in user_input for k in valid_options):
+                errors["base"] = "ambiguous_identifier"
+            elif CONF_REGION_IDENTIFIER in user_input:
+                # Validate region identifier using the API
+                identifier = user_input[CONF_REGION_IDENTIFIER]
 
-            if not errors:
-                # Set the unique ID for this config entry.
-                await self.async_set_unique_id(region_identifier)
-                self._abort_if_unique_id_configured()
+                if not await self.hass.async_add_executor_job(
+                    DwdWeatherWarningsAPI, identifier
+                ):
+                    errors["base"] = "invalid_identifier"
 
-                return self.async_create_entry(title=region_identifier, data=user_input)
+                if not errors:
+                    # Set the unique ID for this config entry.
+                    await self.async_set_unique_id(identifier)
+                    self._abort_if_unique_id_configured()
+
+                    return self.async_create_entry(title=identifier, data=user_input)
+            elif CONF_GPS_TRACKER in user_input:
+                # Validate position using the API
+                device_tracker = user_input[CONF_GPS_TRACKER]
```

Comment:
This is the entity id of the tracker, should we use the entity registry ID instead, which never changes, so this keeps working even if the user changes the name and entity id of the device tracker?

Reply:
Yes, but the entity registry ID is stable and that's what should be stored.

Reply:
Updated in commit 715d339bf96fa18895fc7b0656c2abd5e49b82d5